stream.go

```diff
@@ -264,24 +242,44 @@ func (s *Stream) Reset() error {
 	return nil
 }
 
+func (s *Stream) cancelDeadlines() {
+	s.rDeadline.set(time.Time{})
+	s.wDeadline.set(time.Time{})
+}
+
 func (s *Stream) SetDeadline(t time.Time) error {
-	s.deadlineLock.Lock()
-	defer s.deadlineLock.Unlock()
-	s.rDeadline = t
-	s.wDeadline = t
+	s.clLock.Lock()
+	defer s.clLock.Unlock()
+
+	if s.closedRemote && s.isClosed() {
```

Comment:
this should probably be an or, will fix.

Reply:
done

Reply:
Actually, this should probably be two cases. We want to be able to set a write deadline even when closed for reading and vice versa.

Reply:
It would be inconsistent to set one of the two deadlines I think.